### PR TITLE
Update the benchmark a tiny bit assign only corner/edge once once.

### PR DESCRIPTION
**Pull request by hmaarrfk**
* make sure to fill the array in the setup function. it is unlikely that somebody will pad an empty array. THerefore page faulting won't happen in real usecases.

~~* This also happened to include the modification that will assign every corner or edge only once.~~ I can't actually seem to see any improvement in the benchmark. I removed that code. Though it is immortalized for now in https://github.com/hmaarrfk/numpy/commits/pad_corner_only_once

### CODE REVIEW

**Comment by lagru** [comment]
> I can't actually seem to see any improvement in the benchmark. I removed that code. Though it is immortalized for now in https://github.com/hmaarrfk/numpy/commits/pad_corner_only_once

That's something I noticed as well!

File: benchmarks/benchmarks/bench_lib.py [hunk 1]
```diff
@@ -13,13 +13,29 @@ class Pad(Benchmark):
...
+         (10, 10, 10),
+         # 50 * 512 * 512 = 13 million points = 46 MB. should be a good
+         # out of cache describing a typical usecase
+         (50, 512, 512)],
+        [1,
+         3,
+         (0, 5)],
+        ["constant",
+         "edge", "linear_ramp",
+         # mean/median/minimum/maximum all use the same code path
```
Comment (lagru) [inline benchmarks/benchmarks/bench_lib.py:27]: Nice comments! :)

File: benchmarks/benchmarks/bench_lib.py [hunk 2]
```diff
@@ -13,13 +13,29 @@ class Pad(Benchmark):
...
+         # reflect/symmetric share alot of the code path
+         "reflect",
+         "wrap"],
     ]
 
     def setup(self, shape, pad_width, mode):
         self.array = np.empty(shape)
+        # Make sure to fill the array to make the OS page fault
+        # in the setup phase and not the timed phase
+        self.array.fill(1)
```
Comment (lagru) [inline benchmarks/benchmarks/bench_lib.py]: Could you explain why this is necessary? Or point me to an explanation?

Comment (hmaarrfk) [reply benchmarks/benchmarks/bench_lib.py]: This is important for any allocation larger than basically 4kb. It is really good to keep in so people can change `shape` and see how the benchmark changes.

I found this issue in this comment https://github.com/numpy/numpy/issues/7544#issuecomment-418806035
And opened a formal issue here:
https://github.com/numpy/numpy/issues/11919

You can see the difference in performance when coopying to a preallocated buffer, vs one that was created with `empty`. Page faulting slows down the copy by half. It is really unlikely that somebody will be asking you to pad a `empty` or `zero` array.

And the PR to update some benchmarks in numpy:
https://github.com/numpy/numpy/pull/11959

That PR pointed me to the `full` function, which is cleaner IMO.

**Comment by hmaarrfk** [comment]
> > I can't actually seem to see any improvement in the benchmark. I removed that code. Though it is immortalized for now in https://github.com/hmaarrfk/numpy/commits/pad_corner_only_once
> 
> That's something I noticed as well!

I think this is because asv's definitions of what "improves". We might have to use even larger arrays, to make the delta significant. or change a threshold flag.

**Comment by hmaarrfk** [comment]
Thanks for working on this btw. Improving the performance of numpy for large arrays isn't really trivial, but definitely appreciated!

File: benchmarks/benchmarks/bench_lib.py (hunk 2 above)
Comment (lagru) [reply benchmarks/benchmarks/bench_lib.py]: Thanks!

**Comment by lagru** [comment]
@hmaarrfk Thank you. :)

**Pull request merged**